examples/api/lib/material/text_button/text_button.0.dart

```diff
@@ -6,78 +6,441 @@ import 'package:flutter/material.dart';
-              textStyle: const TextStyle(fontSize: 20),
+    final ThemeData theme = Theme.of(context);
+    final ColorScheme colorScheme = theme.colorScheme;
+
+    // Adapt colors that are not part of the color scheme to
+    // the current dark/light mode. Used to define TextButton #7's
+    // gradients.
+    final (Color color1, Color color2, Color color3) = switch (colorScheme.brightness) {
+      Brightness.light => (Colors.blue.withOpacity(1.0),  Colors.orange.withOpacity(1.0), Colors.yellow.withOpacity(1.0)),
+      Brightness.dark  => (Colors.purple.withOpacity(1.0), Colors.cyan.withOpacity(1.0),  Colors.yellow.withOpacity(1.0)),
+    };
+
+    // This gradient's appearance reflects the button's state.
+    // Always return a gradient decoration so that AnimatedContainer
+    // can interpolorate in between. Used by TextButton #7.
+    Decoration? statesToDecoration(Set<MaterialState> states) {
+      if (states.contains(MaterialState.pressed)) {
+        return BoxDecoration(
+          gradient: LinearGradient(colors: <Color>[color2, color2]), // solid fill
+        );
+      }
+      return BoxDecoration(
+        gradient: LinearGradient(
+          colors: switch (states.contains(MaterialState.hovered)) {
+            true => <Color>[color1, color2],
+            false => <Color>[color2, color1],
+          },
+        ),
+      );
+    }
+
+    return Row(
+      children: <Widget> [
+        // The dark/light and LTR/RTL switches. We use the updateDarkMode function
+        // provided by the parent TextButtonExampleApp to rebuild the MaterialApp
+        // in the appropriate dark/light ThemeMdoe. The directionality of the rest
+        // of the UI is controlled by the Directionality widget below, and the
+        // textDirection local state variable.
+        Card(
+          child: Padding(
+            padding: const EdgeInsets.all(16),
+            child: IntrinsicWidth(
+              child: Column(
+                children: <Widget>[
+                  Row(
+                    children: <Widget>[
+                      const Expanded(child: Text('Dark Mode')),
+                      const SizedBox(width: 4),
+                      Switch(
+                        value: widget.darkMode,
+                        onChanged: (bool value) {
+                          widget.updateDarkMode(value);
+                        },
+                      ),
+                    ],
+                  ),
+                  const SizedBox(height: 16),
+                  Row(
+                    children: <Widget>[
+                      const Expanded(child: Text('RTL Text')),
+                      const SizedBox(width: 4),
+                      Switch(
+                        value: textDirection == TextDirection.rtl,
+                        onChanged: (bool value) {
+                          setState(() {
+                            textDirection = value ? TextDirection.rtl : TextDirection.ltr;
```

Comment:
This controls card takes more vertical space than required. 

We could use  a `BottomAppBar` for controls, see this AppBar [example](https://main-api.flutter.dev/flutter/material/AppBar-class.html#material.AppBar.2).

Reply:
I like the layout ... maybe you mean `horizontal space`? If so, check out the new version. It's packed a little more tightly.

Reply:
Great. Much more compact now.